firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Homepage Rebuild/Redux/PocketMiddlewareTests.swift

```diff
@@ -0,0 +1,82 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Redux
+import XCTest
+
+@testable import Client
+
+final class PocketMiddlewareTests: XCTestCase, StoreTestUtility {
+    let storeUtilityHelper = StoreTestUtilityHelper()
+    let pocketManager = MockPocketManager()
+    override func setUp() {
+        super.setUp()
+        DependencyHelperMock().bootstrapDependencies(injectedPocketManager: pocketManager)
+        setupTestingStore()
+    }
+
+    override func tearDown() {
+        DependencyHelperMock().reset()
+        resetTestingStore()
+        super.tearDown()
+    }
+
+    func test_initializeAction_getPocketData() async throws {
+        let action = HomepageAction(windowUUID: .XCTestDefaultUUID, actionType: HomepageActionType.initialize)
+        store.dispatch(action)
+        try await Task.sleep(nanoseconds: 1_000_000_000)
+
+        let homepageState = store.state.screenState(
+            HomepageState.self,
+            for: .homepage,
+            window: .XCTestDefaultUUID
+        )
+        XCTAssertNotNil(homepageState)
+        XCTAssertEqual(homepageState?.pocketState.pocketData.count, 3)
+        XCTAssertEqual(pocketManager.getPocketItemsCalled, 1)
+    }
+
+    func test_enterForegroundAction_getPocketData() async throws {
+        let action = PocketAction(windowUUID: .XCTestDefaultUUID, actionType: PocketActionType.enteredForeground)
+        store.dispatch(action)
+        try await Task.sleep(nanoseconds: 1_000_000_000)
```

Comment:
not sure how I feel about these sleeps, however, I am unable to think of another way without a larger refactor. If we could mock our store and monitor the dispatches, then we can use an expectation on that dispatch. In our current case our store is a global variable, so can't be mock properly and the alternative was to make it a var so that it can be overridden in tests.

Reply:
@OrlaM I believe you mentioned that we can make a protocol during work week. Could you expand on that idea here?

Reply:
I'm not sure the protocol option will allow us to move away from sleeps but I haven't fully thought it through. I'd say merge this one as is and if you want to tackle the protocol option you can set up a working session with me some time and we can work through it.

Reply:
thanks @OrlaM, I'll go ahead and merge this and set up a working session.